halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentEnvironmentService.java

```diff
@@ -56,14 +62,26 @@ public DeploymentEnvironment getDeploymentEnvironment(String deploymentName) {
 
   public void setDeploymentEnvironment(String deploymentName, DeploymentEnvironment newDeploymentEnvironment) {
     DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+
+    String defaultLocation = lookupDefaultLocation(deploymentName, newDeploymentEnvironment.getAccountName());
+    String location = newDeploymentEnvironment.getLocation();
+    newDeploymentEnvironment.setLocation(StringUtils.isEmpty(location) ? defaultLocation : location);
```

Comment:
@lwander They don't have to repeatedly set it. Once set via CLI, it retains that value until another CLI override changes it. The default only kicks in if (A) the deployment environment doesn't have a location set, and (B) the provider defines a default location.

The case of changing account types from k8s and having it fail because of "spinnaker" is definitely a confusing bit of UX. We mulled that a little and decided to punt it to when we implement the validator.

What would be the "correct" behavior when:
* User sets account type to k8s without setting location (which sets default for that provider)
* User changes account type to openstack

We considered the option of only setting the default location at deploy time, in each micro service (essentially in every place that "spinnaker" was hardcoded before). Maybe that's the best choice after all. The main issue we saw with it was that the default namespace doesn't appear in the config; that feels inconsistent and obscures useful information from the user, but perhaps that's actually the smaller issue here.